hw9/modules/_typesGuardMainClass.ts

```diff
@@ -0,0 +1,8 @@
+//created special type(used type guard) for main classes: tracks, alboms, artists
+//we dont have an unambiguous answer which data type will be used 
+//when declaring a class in a constructor.
+//for type guard we will describe all valid types for any parameters 
+
+type MainClassesTypes = string | number | object | Array<string> | Array<number> | Array<object> | Date;
+//MainClassesTypes includes Generics type : Array<number> and Array<object>
+export type {MainClassesTypes};
```

Comment:
type is not a type guard

Reply:
read more about type guards here:
https://basarat.gitbook.io/typescript/type-system/typeguard#user-defined-type-guards

Reply:
fixed. Thanks for link

Reply:
Add type guards for album, artist, track, library